src/Database/Database.php

```diff
@@ -1461,8 +1461,13 @@ public function count($table, array $condition = [], array $params = [])
 
 		$row = $this->fetchFirst($sql, $condition);
 
-		// Ensure to always return either a "null" or a numeric value
-		return is_numeric($row['count']) ? (int)$row['count'] : $row['count'];
+		if (!isset($row['count'])) {
+			$this->logger->notice('Invalid count.', ['table' => $table, 'expression' => $expression, 'condition' => $condition_string, 'callstack' => System::callstack()]);
+			return 0;
+		} else {
+			// Ensure to always return either a "null" or a numeric value
+			return is_numeric($row['count']) ? (int)$row['count'] : $row['count'];
```

Comment:
The return of `COUNT` cannot be `null`, either the query succeeded and it returns a positive integer, 0 or more, or the query failed for whatever reason and `$row['count']` doesn't exist.

This would be sufficient:
 ```suggestion
			return (int)$row['count'];
```
(Edit: no idea why GitHub adds some spaces for indentation in my suggestion, please keep the tabulations.)